config/configparser/config.go

```diff
@@ -94,16 +93,13 @@ type pipelineSettings struct {
 // typeAndNameSeparator is the separator that is used between type and name in type/name composite keys.
 const typeAndNameSeparator = "/"
 
-// Load loads a Config from Viper.
+// Parse loads a Config from Parser.
 // After loading the config, need to check if it is valid by calling `ValidateConfig`.
-func Load(
-	v *viper.Viper,
-	factories component.Factories,
-) (*configmodels.Config, error) {
+func Parse(v *config.Parser, factories component.Factories) (*configmodels.Config, error) {
```

Comment:
Nit:
```suggestion
func Parse(p *config.Parser, factories component.Factories) (*configmodels.Config, error) {
```

Reply:
Nit: names are a bit confusing. If this is Parse func, then why is it accepting a Parser? Is Parser doing parsing or this function is doing parsing?

Maybe this package should be named configloader instead of configparser and the function is Load instead of Parse?

Reply:
Reverted to `Load`. Will update the package later.